lib/elixir/lib/macro.ex

```diff
@@ -8,14 +8,43 @@ defmodule Macro do
 
   To create a custom sigil, define a function with the name
   `sigil_{identifier}` that takes two arguments. The first argument will be
-  the interpolated string, the second will be a char list containing any
-  modifiers.
+  the string, the second will be a char list containing any modifiers. If the
+  sigil is lower case (such as `sigil_x`) then the string argument will allow
+  interpolation. If the sigil is upper case (such as `sigil_X`) then the string
+  will not be interpolated.
 
   Valid modifiers include only lower and upper case letters. Other characters
   will cause a syntax error.
 
   The module containing the custom sigil must be imported before the sigil
   syntax can be used.
+
+  ### Examples
+
+      defmodule MySigils do
+        defmacro sigil_x(term, [?r]) do
+          quote do: unquote(term) |> String.reverse()
+        end
+        defmacro sigil_x(term, _modifiers) do
+          term
+        end
+        defmacro sigil_X(term, [?r]) do
+          quote do: unquote(term) |> String.reverse()
+        end
+        defmacro sigil_X(term, _modifiers) do
+          term
+        end
+      end
+
+      iex > import MySigils
+      iex > ~x(with #{"inter" <> "polation"})
+      "with interpolation"
+      iex > ~x(with #{"inter" <> "polation"})r
+      "noitalopretni htiw"
+      iex > ~X(without #{"interpolation"})
+      "without \#{"interpolation"}"
+      iex > ~X(without #{"interpolation"})r
```

Comment:
@josevalim I was hoping to be able to doctest them, but I couldn't get it working. I kept getting the following error:

 > module Kernel.MacrosTest.MySigils is not loaded but was defined. This happens when you depend on a module in the same context it is defined. For example:

I have gone with `#=>` for now. I have also squashed the commits.